apps/OpenSpace/ext/launcher/include/profile/assetsdialog.h

```diff
@@ -26,12 +26,39 @@
 #define __OPENSPACE_UI_LAUNCHER___ASSETSDIALOG___H__
 
 #include <QDialog>
-
+#include <QSortFilterProxyModel>
+#include <QRegularExpression>
 #include "assettreemodel.h"
 
 class QTextEdit;
 class QTreeView;
 
+class SearchProxyModel : public QSortFilterProxyModel {
+Q_OBJECT
+public:
+    /**
+     * Constructor for SearchProxyModel class
+     *
+     * \param parent The QObject* object that is the Qt parent of this object
+     */
+    SearchProxyModel(QObject* parent = nullptr) : QSortFilterProxyModel(parent) {}
+
+public slots:
+    /**
+     * Sets the regular expression pattern to apply to the filter
+     *
+     * \param pattern The QString reference containing the regex pattern
+     */
+    void setFilterRegularExpression(const QString& pattern);
+
+protected:
+    bool filterAcceptsRow(int sourceRow, const QModelIndex& sourceParent) const override;
+
+private:
+    QRegularExpression* _regExPattern = nullptr;
+    bool acceptIndex(const QModelIndex& idx) const;
```

Comment:
```suggestion
    bool acceptIndex(const QModelIndex& idx) const;

    QRegularExpression* _regExPattern = nullptr;
```

In general, we sort functions first, then member variables